packages/web-components/src/components/va-modal/va-modal.tsx

```diff
@@ -0,0 +1,310 @@
+    if (!this.isVisibleDirty) return;
+
+    this.isVisibleDirty = false;
+    if (this.visible) {
+      this.setupModal();
+    } else {
+      this.teardownModal();
+    }
+  }
+
+  disconnectedCallback() {
+    this.teardownModal();
+  }
+
+  private handleClose(e) {
+    this.closeEvent.emit(e);
+  }
+
+  /**
+   * This method traps the focus inside our web component, prevents scrolling outside
+   * the modal, and adds aria-hidden="true" to all elements outside the web component.
+   * Fires analytics event unless disableAnalytics is true.
+   */
+  private setupModal() {
+    const initialFocus = (this.el.querySelector(this.initialFocusSelector) ||
+      this.el.shadowRoot.querySelector(
+        this.initialFocusSelector,
+      )) as HTMLElement;
+    this.focusTrap = createFocusTrap(
+      [this.el, this.alertActions, this.contents],
+      {
+        // trap is removed in teardownModal - disable escape key deactivating the focus trap
+        escapeDeactivates: false,
+        // the element we want to target immediately after opening modal
+        initialFocus: initialFocus || this.closeButton,
+      },
+    );
+    this.focusTrap.activate();
+    disableBodyScroll(this.el);
+    this.undoAriaHidden = hideOthers(this.el);
+
+    // NOTE: With this PR (https://github.com/department-of-veterans-affairs/vets-website/pull/11712)
+    // we rely on the existence of `body.modal-open` to determine if a modal is
+    // currently open and adjust programmatic scrolling if there is.
+    document.body.classList.add('modal-open');
+
+    // Conditionally track the event.
+    if (!this.disableAnalytics) {
+      const detail = {
+        componentName: 'va-modal',
+        action: 'show',
+        details: {
+          status: this.status,
+          title: this.modalTitle,
+          primaryButtonText: this.primaryButton?.text,
+          secondayButtonText: this.secondaryButton?.text,
+        },
+      };
+      this.componentLibraryAnalytics.emit(detail);
+    }
+  }
+
+  /**
+   * This method removes the focus trap, re-enables scrolling and
+   * removes aria-hidden="true" from external elements.
+   */
+  private teardownModal() {
+    this.focusTrap?.deactivate();
+    clearAllBodyScrollLocks();
+    this.undoAriaHidden?.();
+
+    document.body.classList.remove('modal-open');
+  }
+
+  render() {
+    const {
+      modalTitle,
+      hideCloseButton,
+      primaryButton,
+      secondaryButton,
+      status,
+      visible,
+    } = this;
+
+    if (!visible) return null;
+
+    const ariaLabel = `${modalTitle} modal` || 'Modal';
+    const wrapperClass = classnames('va-modal-inner', {
+      'va-modal-alert': status,
+      [`va-modal-alert-${status}`]: status,
+    });
+    const bodyClass = status ? 'va-modal-alert-body' : 'va-modal-body';
+    const titleClass = classnames(
+      status ? 'va-modal-alert-title' : 'va-modal-title',
+    );
+    const contentClass = classnames({ 'va-modal-alert-text': status });
```

Comment:
Could we could simplify this `status`-based class logic with something like this in the CSS?

```css


.va-modal-body {
  margin-right: 2rem;
  overflow-wrap: break-word;
  padding: 2rem;
  word-break: break-word;
  word-wrap: break-word;
}

/* the va-modal-alert-body*/
:host([status]) .va-modal-body {
  display: table-cell;
  padding-left: 0;
  padding-right: 0;
  width: 100%;
  vertical-align: middle;
}

```